common/server_protocolhandler.cpp

```diff
@@ -620,23 +620,9 @@ Response::ResponseCode Server_ProtocolHandler::cmdJoinRoom(const Command_JoinRoo
     if (!r)
         return Response::RespNameNotFound;
 
-    QString roomPermission = r->getRoomPermission().toLower();
-    if (roomPermission != "none"){
-        if (roomPermission == "registered") {
-            if (!(userInfo->user_level() & ServerInfo_User::IsRegistered))
-                return Response::RespUserLevelTooLow;
-        }
-
-        if (roomPermission == "moderator"){
-            if (!(userInfo->user_level() & ServerInfo_User::IsModerator))
-                return Response::RespUserLevelTooLow;
-        }
-
-        if (roomPermission == "administrator"){
-            if (!(userInfo->user_level() & ServerInfo_User::IsAdmin))
-                return Response::RespUserLevelTooLow;
-        }
-    }
+    if (!(userInfo->user_level() & ServerInfo_User::IsModerator || userInfo->user_level() & ServerInfo_User::IsModerator))
```

Comment:
This is a double moderator check. You probably want if (NOT ( isAdmin || isModerator ))